arbstate/das_reader.go

```diff
@@ -109,25 +128,29 @@ func DeserializeDASCertFrom(rd io.Reader) (c *DataAvailabilityCertificate, err e
 }
 
 func (c *DataAvailabilityCertificate) SerializeSignableFields() []byte {
-	buf := make([]byte, 0, 32+8)
+	buf := make([]byte, 0, 32+9)
 	buf = append(buf, c.DataHash[:]...)
 
 	var intData [8]byte
 	binary.BigEndian.PutUint64(intData[:], c.Timeout)
 	buf = append(buf, intData[:]...)
 
+	if c.Version != 0 {
+		buf = append(buf, c.Version)
+	}
+
 	return buf
 }
 
 func (cert *DataAvailabilityCertificate) RecoverKeyset(
 	ctx context.Context,
 	da DataAvailabilityReader,
 ) (*DataAvailabilityKeyset, error) {
-	keysetBytes, err := da.GetByHash(ctx, cert.KeysetHash[:])
+	keysetBytes, err := da.GetByHash(ctx, cert.KeysetHash)
 	if err != nil {
 		return nil, err
 	}
-	if !bytes.Equal(crypto.Keccak256(keysetBytes), cert.KeysetHash[:]) {
+	if dastree.Hash(keysetBytes) != cert.KeysetHash {
```

Comment:
What about keysets that were stored before the DASes were upgraded? I suppose we'll just fall back to getting it from the chain so it's probably fine...